pkg/protocol/engine/engine.go

```diff
@@ -556,12 +532,94 @@ func (e *Engine) ErrorHandler(componentName string) func(error) {
 	}
 }
 
-func (e *Engine) RLock() {
-	e.accessMutex.RLock()
+func (e *Engine) initRootCommitment() {
+	updateRootCommitment := func(lastFinalizedSlot iotago.SlotIndex) {
+		e.RootCommitment.Compute(func(rootCommitment *model.Commitment) *model.Commitment {
+			protocolParams := e.APIForSlot(lastFinalizedSlot).ProtocolParameters()
+			maxCommittableAge := protocolParams.MaxCommittableAge()
+
+			targetSlot := protocolParams.GenesisSlot()
+			if lastFinalizedSlot > targetSlot+maxCommittableAge {
+				targetSlot = lastFinalizedSlot - maxCommittableAge
+			}
+
+			if rootCommitment != nil && targetSlot == rootCommitment.Slot() {
+				return rootCommitment
+			}
+
+			return lo.Return1(e.Storage.Commitments().Load(targetSlot))
```

Comment:
Why are you ignoring the error here? In principle this should never happen, but we should still at least log the error

Reply:
we log it now